api.planx.uk/send/bops.ts

```diff
@@ -63,38 +65,49 @@ const sendToBOPS = async (req: Request, res: Response, next: NextFunction) => {
       fixRequestBody(proxyReq, req);
     },
     onProxyRes: responseInterceptor(
-      async (responseBuffer, proxyRes, req, _res) => {
+      async (responseBuffer, proxyRes, req, res) => {
         // Mark session as submitted so that reminder and expiry emails are not triggered
         markSessionAsSubmitted(payload?.sessionId);
 
-        const bopsResponse = JSON.parse(responseBuffer.toString("utf8"));
+        const bopsRawResponse = responseBuffer.toString("utf8");
+
+        let bopsResponse: { id: string } | undefined;
+        try {
+          bopsResponse = JSON.parse(bopsRawResponse);
+        } catch (e) {
+          res.statusCode = 502; // Bad Gateway - invalid response from the upstream server
```

Comment:
This should allow us to interrogate the failed request in hasura and should also let hasura know a retry is required.